packages/shared/src/components/modals/report/ReportUserModal.tsx

```diff
@@ -0,0 +1,117 @@
+import type { ReactElement } from 'react';
+import React, { useState } from 'react';
+import { useMutation } from '@tanstack/react-query';
+import { ReasonSelectionModal } from './ReasonSelectionModal';
+import type { ReportReason } from '../../../report';
+import { ReportEntity, SEND_REPORT_MUTATION } from '../../../report';
+import { Checkbox } from '../../fields/Checkbox';
+import type { UserShortProfile } from '../../../lib/user';
+import { gqlClient } from '../../../graphql/common';
+import {
+  CONTENT_PREFERENCE_BLOCK_MUTATION,
+  ContentPreferenceType,
+} from '../../../graphql/contentPreference';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import { useToastNotification } from '../../../hooks';
+
+const reportReasons: { value: string; label: string }[] = [
+  { value: 'inappropriate', label: 'Inappropriate or NSFW Content' },
+  { value: 'trolling', label: 'Trolling or Disruptive Behavior' },
+  { value: 'harassment', label: 'Harassment or Bullying' },
+  { value: 'impersonation', label: 'Impersonation or False Identity' },
+  { value: 'spam', label: 'Spam or Unsolicited Advertising' },
+  { value: 'misinformation', label: 'Misinformation or False Claims' },
+  { value: 'hateSpeech', label: 'Hate Speech or Discrimination' },
+  { value: 'privacy', label: 'Privacy or Copyright Violation' },
+  { value: 'plagiarism', label: 'Plagiarism or Content Theft' },
+  { value: 'other', label: 'Other' },
+];
+
+type ReportUserModalProps = {
+  offendingUser: UserShortProfile;
```

Comment:
Maybe changing to `Pick<UserShortProfile, 'id' | 'username'>` would be better because we don't need all the infos (and we don't have in post for example)